### ISSUE TITLE
PowerPC support

### ISSUE THREAD

**Issue by Rowan-Bird** [pmj/QemuUSBTablet-OSX#6]
Can you please make a PowerPC version because QEMU's awful mouse emulation is unbearable on Mac OS X 10.2

**Comment by rocheston** [comment]
I too want this. For Mac OS X 10.2. Please create a PowerPC version.

**Comment by pmj** [comment]
Have you tried building the existing code with an Xcode version which still contains a PPC compiler and SDK? Also, I've not run a PPC MacOSX for years, any pointers for getting me up and running with a test system as quickly as possible will increase the chances I'll end up giving this a go.

**Comment by pmj** [comment]
Incidentally, on x86 with modern macOS versions on vaguely recent Qemu versions, this driver is no longer needed, as I managed to get a patch that fixes Qemu's Tablet to be standards compliant and correctly driven by macOS's builtin driver. I'm guessing this isn't working correctly on PPC?

**Comment by rocheston** [comment]
Nope. I get this error Materials are  missing....bla..bla.. when I execute the program in Mac OS X 10.2. Does not install. Guess PowerPC.

**Comment by pmj** [comment]
Is that when attempting to open the installer .pkg? If so, try manually placing the codeless kext in `/System/Library/Extensions`:

* Download Build-1.2.zip from https://github.com/pmj/QemuUSBTablet-OSX/tree/master/build
* Extract it
* Install the kext, using something along the lines of: `sudo cp -r Build-1.2/artefacts/QemuUSBTablet/System/Library/Extensions/QemuUSBTablet.kext /System/Library/Extensions/`
* Reboot

And see what happens. If it doesn't work, please provide:

- Qemu command line used (especially the USB tablet part, but it's hard to say in advance what's relevant and what isn't, so to avoid being led on a wild goose chase, the whole thing would be best.)
- What sort of host system are you running this on?
- What exactly goes wrong? What are you expecting? What actually happens?
- What have you tried to fix the problem?
- Run `ioreg -lirc IOUSBDevice` from the Terminal inside the VM` and attach its output here (I'm only interested in HID devices, but if you're not sure, just attach all of it, I can easily find what I need)

**Comment by rocheston** [comment]
YES! Your solution worked like a charm. I moved the kext file to /System/Library/Extensions and rebooted. The mouse is super smooth now. Thank you for this solution.

Cheers!
